### PR TITLE
gh-87452 Making returncode unreliability explicit

### PR DESCRIPTION
**Pull request by zitterbewegung**
We make explicit that the value stored in return code is best returned by poll instead of using the value in return code. 

 gh-87452: Making returncode unreliability explicit

### CODE REVIEW

**Comment by benjaminp** [comment]
Thank you for the PR.

I don't think "out-of-date" is the right term here. Either the entire `subprocess.Subprocess` instance is aware of the exit of the child process and knows the return code, or it doesn't. The first paragraph of the method documentation could perhaps be rearranged to more clearly state the order of events instead of appending another sentence.

**Comment by benjaminp** [comment]
I would propose:
```
The child return code. Initially ``None``, :attr:`returncode` is set by :meth:`poll` and :meth:`wait` after process termination is observed.
``

File: Doc/library/subprocess.rst [hunk 1]
```diff
@@ -890,14 +890,16 @@ The following attributes are also available:
 
 .. attribute:: Popen.returncode
 
-   The child return code, set by :meth:`poll` and :meth:`wait` (and indirectly
-   by :meth:`communicate`).  A ``None`` value indicates that the process
+   The child return code. The child return code. Initially ``None``, :attr:`returncode` is set by :meth:`poll` and :meth:`wait` after process termination is observed.
+   
+   A ``None`` value indicates that the process
    hasn't terminated yet.
```
Comment (benjaminp) [inline Doc/library/subprocess.rst]: We can delete this sentence now.

File: Doc/library/subprocess.rst [hunk 2]
```diff
@@ -890,14 +890,16 @@ The following attributes are also available:
...
 
    A negative value ``-N`` indicates that the child was terminated by signal
    ``N`` (POSIX only).
 
 
+
```
Comment (benjaminp) [inline Doc/library/subprocess.rst]: and this newline

File: Doc/library/subprocess.rst (hunk 1 above)
Comment (zitterbewegung) [reply Doc/library/subprocess.rst]: I will remove the sentence and also remove the duplicate sentence "The child return code."

**Comment by erlend-aasland** [comment]
Please remove the duplicate NEWS entry.

File: Doc/library/subprocess.rst [hunk 3]
```diff
@@ -894,14 +894,16 @@ The following attributes are also available:
 
 .. attribute:: Popen.returncode
 
-   The child return code, set by :meth:`poll` and :meth:`wait` (and indirectly
-   by :meth:`communicate`).  A ``None`` value indicates that the process
+   The child return code. The child return code. Initially ``None``, :attr:`returncode` is set by :meth:`poll` and :meth:`wait` after process termination is observed.
```
Comment (slateny) [inline Doc/library/subprocess.rst]: The docs failure is probably due to this line being too long - can you wrap it and keep it no more than 80 chars?

Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   The child return code. Initially ``None``, :attr:`returncode` is set by :meth:`poll` and :meth:`wait` after process termination is observed.
```

File: Doc/library/subprocess.rst [hunk 4]
```diff
@@ -894,14 +894,16 @@ The following attributes are also available:
 
 .. attribute:: Popen.returncode
 
-   The child return code, set by :meth:`poll` and :meth:`wait` (and indirectly
-   by :meth:`communicate`).  A ``None`` value indicates that the process
+   The child return code. Initially ``None``, :attr:`returncode` is set by :meth:`poll` and :meth:`wait` after process termination is observed.
```
Comment (erlend-aasland) [inline Doc/library/subprocess.rst]: Please wrap at 79 chars

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
Again, please remove the duplicate NEWS entry. Also please address Benjamin's remarks.

**Comment by slateny** [comment]
@zitterbewegung Would you still be interested in giving the PR a quick update per the reviews above?

**Comment by zitterbewegung** [comment]
Sorry I can do that.

File: Doc/library/subprocess.rst (hunk 1 above)
Comment (zitterbewegung) [reply Doc/library/subprocess.rst]: Fixed

File: Doc/library/subprocess.rst (hunk 2 above)
Comment (zitterbewegung) [reply Doc/library/subprocess.rst]: Fixed

File: Doc/library/subprocess.rst (hunk 3 above)
Comment (zitterbewegung) [reply Doc/library/subprocess.rst]: I have addressed this issue.

**Review by zitterbewegung — COMMENTED** [review]
- Fixed word wrap issue
- Removed duplicate news entry.

**Review by zitterbewegung — COMMENTED** [review]
Removed trailing whitespace

**Review by furkanonder — APPROVED** [review]

**Comment by zitterbewegung** [comment]
@erlend-aasland can this be reviewed?

**Comment by erlend-aasland** [comment]
Sorry for the delay; thanks Greg for merging. Thanks for the PR, Joshua.

**Pull request merged**